### DISCUSSION TITLE
API-Request fail (Error: 400) - what is wrong with it?

### DISCUSSION THREAD

**Discussion by xstable** [kimai/kimai#4274] [Questions & Answers]
I've executed the API-Form you can access via /api/doc.
I've tried to get all time entries for a given Project.

![image](https://github.com/kimai/kimai/assets/1778068/65012fb4-9ab9-41c3-bfe8-ece9a3020dbb)

Any Idea, why this error was thrown?

I assume it's depends somehow at the date? I've also tried this Format: 2023-08-01...

**Comment by kevinpapst** [comment]
No log, no answer. Wild guess: Invalid datetime string.
Next time, please copy & paste the entire curl call, so it is possible to retry with another system.

**Comment by xstable** [comment]
I din't do any curl at my own, I'd only used the doc-page as you see in the screenshot. So don't see what you mean with paste the entire call. Isn't it in the `curl` window of my screenshot?

If I keep the dates for begin / end empty, the request succeed. So maybe a bug in this doc-form?

**Reply by kevinpapst** [reply]
> Isn't it in the curl window of my screenshot?

Yes. But do you really believe I am going to copy that character by character?
Next time copy & paste that here, so I can try and tell you what is wrong.

Looking at it, the date format looks wrong to me. Maybe the UI doesn't format it correctly. 
Try something like `2023-07-03T00:00:00`

**Comment by xstable** [comment]
Tried it as you suggest, with same error. 
Here is the copied curl:

```
curl -X 'GET' \
  'https://xxxx.com/api/timesheets?user=1&project=1&begin=2023-08-01T00%3A00%3A00&end=2023-08-31T00%3A00%3A00' \
  -H 'accept: application/json' \
  -H 'X-AUTH-USER: xxxx' \
  -H 'X-AUTH-TOKEN: xxxx'
  ```

**Reply by kevinpapst** [reply]
Try one new value at a time. 
Remove user/project/end. 
Retry. 
Add one value a a time.
Let me know what you find out.

**Reply by xstable** [reply]
Ok, that's strange, I tried it today with the exact same curl command I've posted above. Today I got a Response of 200 and some data. 
Don't know where the issue was here.
Are there any logs which record those api-calls?

**Reply by kevinpapst** [reply]
In "dev" mode all requests will be logged. In production, you might see the failures in var/logs/prod.log
